### PR TITLE
Add TypeScript for server code

### PR DESCRIPTION
**Pull request by olafkrawczyk**
* Add TypeScript packages
 * Create interfaces for existing objects
 * Use ts-node to start applicaion

Resolves ISSUE-29

---
This change is [Reviewable](https://reviewable.io/reviews/piotrmajcher94/poker-app/43)

### CODE REVIEW

File: poker-app-server/src/interfaces/deck-interface.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { Card } from "./card-interface";
+
+export interface Deck {
```
Comment (piotrmajcher94) [inline poker-app-server/src/interfaces/deck-interface.ts]: I think we don't really need the  Deck interface. It's a legacy from when we stored the deck in json file, as we had to import an object from a json file, we could not import an array directly. I am pretty sure we can just export an array of Cards now.

File: poker-app-server/src/resources/deck-of-cards.ts [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+      value: CardValue.king
+    },
+    {
+      suit: Suit.spades,
+      value: CardValue.ace
+    }
+  ]
+}
+
+export default deck;
```
Comment (piotrmajcher94) [inline poker-app-server/src/resources/deck-of-cards.ts]: We can probably just export Card[] here => _I think we don't really need the  Deck interface. It's a legacy from when we stored the deck in json file, as we had to import an object from a json file, we could not import an array directly. I am pretty sure we can just export an array of Cards now._

File: poker-app-server/test/services/card-deck-service.spec.ts [hunk 3]
```diff
@@ -226,6 +226,7 @@ describe('CardDeckService Tests', function() {
     expect(actualDeck).to.exist;
     expect(expectedDeck).not.to.deep.equal(actualDeck);
     expect(EXPECTED_DECK_LENGTH).to.equal(actualDeck.length);
-    expect(expectedDeck).to.have.deep.all.members(actualDeck);
+    expect(expectedDeck).to.have.deep.members(actualDeck);
+    expect(expectedDeck).to.have.deep.members(actualDeck);
```
Comment (piotrmajcher94) [inline poker-app-server/test/services/card-deck-service.spec.ts]: Same line repeated twice. Why did you erase .all ?

File: poker-app-server/src/repositories/game-repository.ts [hunk 4]
```diff
@@ -1,54 +1,55 @@
...
     /**
      * Updates the game state
-     * 
+     *
      * @param {Object} newVersion
      * @returns {Object} updated game state object
      */
-    update(newVersion = {}) {
+    update(newVersion: Game): Game {
         const id = newVersion.id;
```
[the remark this answers is not in the document]
Comment (olafkrawczyk) [reply poker-app-server/src/repositories/game-repository.ts]: Let's assume that!

**Pull request merged**